hack/lib.sh

```diff
@@ -128,43 +128,44 @@ function enable_workload_identity(){
+    --workload-pool="${project_id}".svc.id.goog
 
   # Modify all node pools to enable GKE_METADATA.
    echo "Modify all node pools to enable GKE_METADATA"
-  pools=$(gcloud container node-pools list --cluster=${cluster_name} --${cluster_location_option}=${cluster_location} --format="value(name)")
+  pools=$(gcloud container node-pools list --cluster="${cluster_name}" --${cluster_location_option}="${cluster_location}" --format="value(name)")
   while read -r pool_name
   do
   gcloud container node-pools update "${pool_name}" \
-    --cluster=${cluster_name} \
-    --${cluster_location_option}=${cluster_location} \
+    --cluster="${cluster_name}" \
+    --${cluster_location_option}="${cluster_location}" \
     --workload-metadata=GKE_METADATA
   done <<<"${pools}"
 
-  gcloud projects add-iam-policy-binding ${project_id} \
-    --member=serviceAccount:${control_plane_service_account}@${project_id}.iam.gserviceaccount.com \
+  gcloud projects add-iam-policy-binding "${project_id}" \
+    --member=serviceAccount:"${control_plane_service_account}"@"${project_id}".iam.gserviceaccount.com \
     --role roles/iam.serviceAccountAdmin
   }
 
 function storage_admin_set_up() {
   echo "Update ServiceAccount for Storage Admin"
   local project_id=${1}
   local pubsub_service_account=${2}
-  local pubsub_service_account_key_temp=${3}
+  local project_number="$(gcloud projects describe "${project_id}" --format="value(projectNumber)")"
+  local access_token="$(gcloud auth application-default print-access-token)"
 
   echo "parameter project_id used when setting up storage admin is'${project_id}'"
   echo "parameter pubsub_service_account used when setting up storage admin is'${pubsub_service_account}'"
-  echo "parameter pubsub_service_account_key_temp used when setting up storage admin is'${pubsub_service_account_key_temp}'"
   echo "Update ServiceAccount for Storage Admin"
   gcloud services enable storage-component.googleapis.com
   gcloud services enable storage-api.googleapis.com
-  gcloud projects add-iam-policy-binding ${project_id} \
-    --member=serviceAccount:${pubsub_service_account}@${project_id}.iam.gserviceaccount.com \
+  gcloud projects add-iam-policy-binding "${project_id}" \
+    --member=serviceAccount:"${pubsub_service_account}"@"${project_id}".iam.gserviceaccount.com \
     --role roles/storage.admin
-  export GCS_SERVICE_ACCOUNT=`curl -s -X GET -H "Authorization: Bearer \`GOOGLE_APPLICATION_CREDENTIALS=${pubsub_service_account_key_temp} gcloud auth application-default print-access-token\`" "https://www.googleapis.com/storage/v1/projects/${project_id}/serviceAccount" | grep email_address | cut -d '"' -f 4`
-  gcloud projects add-iam-policy-binding ${project_id} \
-    --member=serviceAccount:"${GCS_SERVICE_ACCOUNT}" \
+
+  curl -s -X GET -H "Authorization: Bearer ${access_token}" "https://www.googleapis.com/storage/v1/projects/${project_id}/serviceAccount"
+  gcloud projects add-iam-policy-binding "${project_id}" \
```

Comment:
We need to update the docs accordingly https://github.com/google/knative-gcp/blob/master/docs/examples/cloudstoragesource/README.md

Reply:
The doc does not need to be updated. Here it's using Option 1 to get the GCS service account, except instead of visiting the pantheon UI, it's using the curl command to activate the service account.